extension/js/common/core/crypto/smime/smime-key.ts

```diff
@@ -109,24 +88,28 @@ export class SmimeKey {
   }
 
   private static getKeyFromCertificate = (certificate: forge.pki.Certificate, pem: string): Key => {
+    if (!certificate.publicKey) {
+      throw new UnreportableError(`This S/MIME x.509 certificate doesn't have a public key`);
+    }
+    const fingerprint = forge.pki.getPublicKeyFingerprint(certificate.publicKey, { encoding: 'hex' }).toUpperCase();
```

Comment:
Not for this issue, but would be nice to write tests to make sure we can also work with ECC x509 keys, like curve25519, P256, P512